kafka/kafka.sh

```diff
@@ -24,49 +30,73 @@ function update_apt_get() {
   return 1
 }
 
-update_apt_get
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
 
-# Only run the installation on workers; verify zookeeper on master(s)
-if [[ $(/usr/share/google/get_metadata_value attributes/dataproc-role) == 'Master' ]]; then
-  service zookeeper-server status || \
-      (echo 'Required zookeeper-server not running on master!' && exit 1)
-  # On master nodes, just install kafka libs but not kafka-server.
-  apt-get install -y kafka
-  exit 0
-fi
+function install_and_configure_kafka_server() {
+  # Find zookeeper list first, before attempting any installation.
+  local ZOOKEEPER_CLIENT_PORT=$(grep 'clientPort' /etc/zookeeper/conf/zoo.cfg \
```

Comment:
One more small note about naming: local variables should be lower cased.

Reply:
Ok, I applied to all local variables.